driver/config/config.go

```diff
@@ -228,7 +228,8 @@ type (
 	}
 	SelfServiceStrategyCode struct {
 		*SelfServiceStrategy
-		PasswordlessEnabled bool `json:"passwordless_enabled"`
+		PasswordlessEnabled                bool `json:"passwordless_enabled"`
+		PasswordlessLoginWithAnyCredential bool `json:"passwordless_login_with_any_credential"`
```

Comment:
i mean right now you still need to register an account, the login flow doesn't auto create accounts (except in oidc flows). We could adopt this strategy in future here as well, but this PR addresses when a user already has an account on a different credential, we could allow users to login with a one-time code even if they only have an account registered through password or oidc. 

making this configurable adds flexibility instead of just forcing one way or the other.

Reply:
That's not what I mean. I just don't see a use case for only letting users login after they set up the code method for their account. Either the admin wants to allow login via code or they don't. IMO, this config option requires understanding of how we implemented this and doesn't really add value.

Another config option just adds complexity to the code and for users implementing Ory. We already have 100+ config options, that can be confusing to newcomers. And in this specific case, I just don't see the point of disabling it.

Reply:
I also might be misunderstanding this.

IIUC, when enabled, this option allows users to login via code, even if we didn't store a credential in the `identity_credentials` table and use _any_ of the credentials stored in that table, even if it's _not_ for the `code` type. Is that correct?

If so, see above.

Reply:
> IIUC, when enabled, this option allows users to login via code, even if we didn't store a credential in the `identity_credentials` table and use _any_ of the credentials stored in that table, even if it's _not_ for the `code` type. Is that correct?

That's correct and the discussion is warranted. Both sides have - IMO - valid arguments. If this PR is enabled per default, the login with code method is essentially the same as account recovery, because it allows someone to sign in even if they lost access to their credentials.

**Feature flag**
- Pro: By explicitly enabling this feature, the administrator has control over whether they accept a weaker / different type of proving authenticity than was originally set up by the user (e.g. using code method to "work around" a lost passkey). 
- Contra: Another config option and questionable UX

**Enable per default**
- Pro: One config option less, one way of doing it
- Contra: Login with code now essentially is a recovery flow. I can recover access to my account if I lost access to may passkeys.

I think the UX issues can be addressed by showing the user only the credentials they have set up for login.

Reply:
I'm not sure i follow 100% @jonas-jonas. I think the Admin needs to control if they allow Email OTP logins (and therefore also if it shows as an option in AX), or not; indeed, enabling it makes account recovery via emails kinda obsolete, but 🤷 

IMHO we should have this flag and not force it onto projects, but we should name it more clearly to convey it "enables Email Code login for all users". "any credential type" is somewhat confusing :)

Reply:
I just don't get why you'd want to only allow signing in with a code if the user set it up. How would they even do it in this example scenario where this setting is disabled, but login via code is enabled: A user registers with a password. Now they want to log in with the code via email. How, e.g. what steps does the user need to take to do it?

With this option enabled, it's pretty easy: put in the email address and click "login via code".

But with `passwordless_login_with_any_credential` set to `false` on a project level: ???

And sure, it's essentially the recovery flow now, and in an effort to clean up the code base, we should probably re-use a lot of the code. But I don't see the issue with that. The recovery flow still has its own templates, and redirects to settings after. So it still serves its own purpose...

Reply:
> I just don't get why you'd want to only allow signing in with a code if the user set it up. How would they even do it in this example scenario where this setting is disabled, but login via code is enabled: A user registers with a password. Now they want to log in with the code via email. How, e.g. what steps does the user need to take to do it?

You wouldn't be able to do that which is the point of this feature flag. It's true that you currently can't set up login with code after you have signed up. We initially wanted to add this to the credentials section in the settings screen but so far have not.